l2geth/core/vm/evm.go

```diff
@@ -135,6 +135,24 @@ func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, err
 			}
 			return callStateManager(input, evm, contract)
 		}
+
+		// When the address manager is called
+		if contract.Address() == AddressManagerAddress {
+			// If the first four bytes match `isDeployerAllowed(address)`
+			if bytes.Equal(input[0:4], isDeployerAllowedSig) {
+				// Only in the case where EnableArbitraryContractDeployment is
+				// set, this allows this codepath to be skipped when it is not
+				// set
+				if EnableArbitraryContractDeployment != nil {
```

Comment:
Might be marginally cheaper to put this if statement on the outside instead of the `contract.Address()` check.